neuralop/layers/padding.py

```diff
@@ -39,8 +40,12 @@ def pad(self, x):
         """
         resolution = x.shape[2:]
 
+        # if domain_padding is list, then to pass on
         if isinstance(self.domain_padding, (float, int)):
             self.domain_padding = [float(self.domain_padding)]*len(resolution)
+
+        assert len(self.domain_padding) == len(resolution)
```

Comment:
Good check to have! Can you add a message if that's not the case to explain the issue to users?

Reply:
Explanation is added. "domain_padding length must match the number of spatial/time dimensions (excluding batch, ch)"